README.md

```diff
@@ -759,802 +565,169 @@ If no information is available, this field SHOULD be set to `{}`.
-
-**NB: a Receipt does not guarantee correctness of the result!** The statement's veracity MUST be only understood as an attestation from the executor.
-
-Receipts MUST use the same version as the invocation that they contain.
-
-## 8.1 Schema
-
-```ipldsch
-type Receipt struct {
-  ran     &Invocation
-
-  # output of the invocation
-  out     Result
-
-  # Effects to be performed
-  fx      Effects
-
-  # All the other metadata
-  meta    {String: any}
-
-  # Principal that issued this receipt.
-  # If omitted issuer is inferred from 
-  # the invocation task audience.
-  iss     optional Principal
-
-  # When issuer is different from executor
-  # this MUST hold a UCAN delegation chain 
-  # from executor to the issuer.
-  # This field SHOULD be omitted executor is an issuer.
-  prf     [&UCAN]
-
-  # Signature from the `iss`.
-  s       Varsig
-}
-```
-
-## 8.2 Fields
-
-### 8.2.1 Ran Invocation
-
-The `ran` field MUST include a link to the [Invocation] that the Receipt is for.
-
-### 8.2.2 Output
-
-The `out` field MUST contain the value output of the invocation in [Result] format.
-
-### 8.2.3 Effect
-
-The OPTIONAL `fx` field, if present MUST be set to the caused [Effect]. The [Executor] SHOULD invoke contained [Task] to progress a workflow execution.
-
-If `fx` does not contain OPTIONAL `join` field, it denotes completion of the current execution thread.
-
-### 8.2.4 Metadata Fields
-
-The OPTIONAL metadata field MAY be omitted or used to contain additional data about the receipt. This field MAY be used for tags, commentary, trace information, and so on.
-
-### 8.2.5 Receipt Issuer
-
-The OPTIONAL `iss` field, if present MUST contain the DID of the [Executor] delegate that signed it. If field is present, delegation from [Executor] MUST be included in the `prf` field.
+### 4.2.2 Enqueue
 
-If `iss` field is omitted, Receipt MUST be signed by the [Executor].
+The result of an [Invocation] MAY include a request for further actions to be performed. This is a process of requesting that the invoker "enqueue" a new Task. This enables several things: a clean separation of pure return values from requesting impure tasks to be performed by the runtime, and gives the runtime the control to decide how (or if!) more work should be performed.
 
-### 8.2.6 Proofs
+Enqueued [Task]s describe requests for future work to be performed. They SHOULD come with [Delegation]s, but MAY be a simple request back to the Invoker.
 
-If OPTIONAL `prf` field is present, MUST contain link to UCAN delegation authorizing Receipt Issuer (`iss`) to carry [Task] execution.
+All [Task]s in an [enqueue] array MUST be treated as concurrent, unless explicit data dependencies between them exist via [UCAN Promise]s.
```

Comment:
> For what it's worth join was analogy to a [thread.join](https://doc.rust-lang.org/stable/std/thread/struct.JoinHandle.html#method.join)

Right, but that's not how `join` works. You don't chain `join`s together to continue computation. The analogy seems like a stretch to me.

> but here we lost something that some of our work started to depend on.

We can add back in a special field if there's a clear use case, but can you explain why you can't use the 0-index of the array.

At least _structurally_, these are equivalent:

```haskell
data Fx = [Task]

data Fx' = Fx' {
  cont :: Maybe Task
  fork :: [Task]
}
```

Since this is structured more like actors or dataflow, there isn't really a "main" thread, right?

Reply:
> You don't chain joins together to continue computation.

That is precisely what we do in workflows that process data through multiple queues. We usually have `piece/offer` task that will have .join on `piece/accept` which in turn will have `join` on `aggregate/offer`  etc...

> can you explain why you can't use the 0-index of the array.

If it's an array with one element does it imply that it is a continuation or concurrent task ? Whatever's answer how do I express case where I have not both but the other from one you've answered.

In other words we wan to express tree different cases `{fork}`, `{join}` and `{fork, join}` with `[Task, ...Task[]]`  we only get `[Join]`  or `[Join, ...Fork[]]` and can not get `[...Fork[]]` unless we add another signal (which is what the current structure did)


----

I suspect for IPVM `join` is essentially inferred from `await/...`'s, which makes sense. However in our experience it is a lot easier to UCAN-ify legacy systems if you can simply express `join`'s without having to invert the control flow.